README.md

```diff
@@ -917,39 +921,89 @@ end
 
-Obviously at the first run it will reset everything because it needs to lock all the index specifications in the `Chewy::Stash`.
+```bash
+rake chewy:update # updates all the existing indices
+rake chewy:update[users] # updates UsersIndex only
+rake chewy:update[users,places#city] # updates the whole UsersIndex and PlacesIndex::City type
+rake chewy:update[-users,places#city] # updates every index in application except all the types defined in UsersIndex and the rest of the types defined in PlacesIndex
+```
 
-See [Chewy::Stash](lib/chewy/stash.rb) and [Chewy::Index::Specification](lib/chewy/index/specification.rb) for more details.
+#### `chewy:sync`
+
+Provides a way to synchronize outdated indexes with the source quickly and without doing a full reset.
 
-In the future, additional routines are planned during `chewy:deploy` execution. Like, additional fast or partial index updates to make sure everything is up-to-date as it was a full reset.
+Arguments are similar to the `chewy:update` ones. It is possible to specify a particular type or a whole index.
 
-Right now the approach is that if some data had been updated, but index specification had not been changed, it would be much faster to perform manual partial index update inside data migrations or even manually after the deploy.
+See [Chewy::Type::Syncer](lib/chewy/type/syncer.rb) for more details.
+
+```bash
+rake chewy:sync # synchronizes all the existing indices
+rake chewy:sync[users] # synchronizes UsersIndex only
+rake chewy:sync[users,places#city] # synchronizes the whole UsersIndex and PlacesIndex::City type
+rake chewy:sync[-users,places#city] # synchronizes every index in application except all the types defined in UsersIndex and the rest of the types defined in PlacesIndex
+```
+
+#### `chewy:deploy`
+
+This rake task is especially useful during the production deploy. It is a combination of `chewy:upgrade` and `chewy:sync` and the latter one is called only for the indexes that were not reset during the first stage.
+
+It is not possible to specify any particular types/indexes for this task as it doesn't make much sense.
+
+Right now the approach is that if some data had been updated, but index definition had not been changed no changes satisfying the synchronization algorithm were done, it would be much faster to perform manual partial index update inside data migrations or even manually after the deploy.
 
 Also, there is always full reset alternative with `rake chewy:reset`.
 
+##### Parallelizing rake tasks
+
+Every task described above has its own parallel version. Every parallel rake task takes the number for processes for execution as the first argument and the rest of the arguments are exactly the same as for the non-parallel task version.
+
+To use this task it is require to install [https://github.com/grosser/parallel](https://github.com/grosser/parallel) gem.
+
+If the number of processes is not specified - `parallel` gem tries to derive it from the system information.
```

Comment:
`parallel` gem tries to automatically derive the number of processes to run when it is not explicitly defined